### ISSUE TITLE
Procedure to use Nexus with launchpad mission

### ISSUE THREAD

**Issue by cmoulliard** [openshiftio/appdev-documentation#224]
Here is a procedure describing how to setup Nexus and using OCP 1.6 assign an ENV var to pass the MAVEN_MIRROR_URL to be used to fetch the artifacts using Nexus

Remark : You can use launchpad to create a demo project instead of using the steps oc create is,  oc create bc and oc new-app --template=jenkins-persistent as described hereafter

```
# Use Minishift 1.0.0.RC

minishift delete
minishift config set memory 4096
minishift config set openshift-version v3.6.0-alpha.1
minishift start

oc login -u developer -p developer

eval $(minishift docker-env)
docker pull openshift/jenkins-2-centos7
docker pull openshiftio/launchpad-jenkins-slave
docker pull sonatype/nexus

echo "Setup Nexus application"
oc new-project infra
oc new-app sonatype/nexus
oc expose svc/nexus
oc volumes dc/nexus --add --name 'nexus-volume-1' --type 'pvc' --mount-path '/sonatype-work/' --claim-name 'nexus-pv' --claim-size '1G' --overwrite

echo "Create demo project containing Jenkins Pipeline & Jenkins Server"
echo "Replace within the following BuildConfig the Github project that you want to use AND http address of your Nexus Server"
oc new-project demo
echo "See jenkins centos docker doc - script kubernetes shell"
#oc policy add-role-to-user edit system:serviceaccount:demo:default
#oc policy add-role-to-user admin system:serviceaccount:demo:jenkins

echo "Create ImageStream"
cat << EOF | oc create -f - 
apiVersion: v1
kind: ImageStream
metadata:
  name: launchpad-maven
  annotations:
    openshift.io/display-name: 'Jenkins 2 CentOS 7'
  labels:
    role: jenkins-slave
spec:
  dockerImageRepository: openshiftio/launchpad-jenkins-slave
  tags:
  - from:
      kind: DockerImage
      name: openshiftio/launchpad-jenkins-slave:latest
    name: latest
    referencePolicy:
      type: Source
EOF

oc new-app --template=jenkins-persistent -n demo

echo "Edit Jenkins Conf System to define for launch-maven kubernetes pod template this ENV var"
Key: MAVEN_MIRROR_URL
Val: ${MAVEN_MIRROR_URL}

https://www.dropbox.com/s/a0nxmew0igo72ee/Screenshot%202017-05-08%2017.48.59.png?dl=0

echo #Trigger a new Jenkins Deployment to reload the modification other jenkins build will hang. To be investigated !!!"
oc rollout latest dc/jenkins

echo "Create Buildconfig"
cat << EOF | oc create -f - 
apiVersion: v1
kind: BuildConfig
metadata:
  annotations:
    pipeline.alpha.openshift.io/uses: '[{"name": "jenkins", "namespace": "", "kind":
      "DeploymentConfig"}]'
  name: demo
spec:
  nodeSelector: {}
  output: {}
  postCommit: {}
  resources: {}
  runPolicy: Serial
  source:
    git:
      ref: master
      uri: https://github.com/cmoulliard/zz1.git
    type: Git
  strategy:
    type: JenkinsPipeline
    jenkinsPipelineStrategy:
      jenkinsfilePath: Jenkinsfile
      env:
        - name: MAVEN_MIRROR_URL
          value: http://nexus-infra.192.168.64.25.nip.io/nexus/content/groups/public/
  triggers:
  - type: ConfigChange
  - github:
      secret: "1494236915073"
    type: GitHub
EOF
```

**Comment by inoxx03** [comment]
This procedure does not work on CDK version 3.0.0: when I tested it using CDK, there was an authentication issue while logging into the Jenkins instance.
Also, the `infra` project containing the Nexus instance did not show up on the web console when created, but it did show up using `oc get projects` on the terminal.

**Comment by inoxx03** [comment]
please see PR #245

**Comment by inoxx03** [comment]
For reference: https://docs.openshift.com/container-platform/3.4/dev_guide/app_tutorials/maven_tutorial.html#nexus-setting-up-nexus

**Comment by inoxx03** [comment]
fixed in PR #245
